src/main/java/mate/academy/springbootintro/repository/book/BookRepository.java

```diff
@@ -3,8 +3,14 @@
 import mate.academy.springbootintro.model.Book;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
+import java.util.List;
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+
+    @Query("SELECT b FROM Book b LEFT JOIN FETCH b.categories c"

```

Comment:
No need for nested SELECT - try `SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :categoryId"`